docs/user/precheck/check_networks.rst

```diff
@@ -0,0 +1,199 @@
+=================================
+Check Neutron objects overlapping
+=================================
+
+One of the most tricky parts of Cloud migration is :ref:`networking`. During
+migration to the non-empty cloud there can be plenty of conflicts and overlaps
+from the Neutron side. And it is better to know such issues before the actual
+migration starts. For detecting this kind of problems :dfn:`CheckNetworks`
+action exists.
+
+This action is defined as ``check_networks`` task in the ``preparation``
+section of :ref:`scenario <scenario-files-config>`. There are several examples
+of migration scenarios with this action can be found in the
+:file:`CloudFerry/scenario` folder, like :file:`cold_migrate.yaml` (all-in-one
+migration scenario, tries to migrate all the openstack objects in a single run)
+or :file:`stages/0_prechecks.yaml` (runs all initial checks, which allows to
+spot most of the problems in configuration).
+
+.. important::
+
+    It is highly recommended to run this check before every :ref:`networking`
+    to detect any possible conflicts and resolve it before actual migration
+    starts.
+
+Currently CheckNetworks action contains the following checks:
+
+ - External networks mapping
+ - External networks overlap
+ - Networks segmentation IDs overlap
+ - Floating IPs overlap
+ - Busy physical networks on DST of FLAT network type
+ - Physical networks existence on DST for VLAN network type
+ - VMs spawned directly in external network
+
+
+External networks mapping
+-------------------------
+
+Checks :ref:`external networks mapping <external-networks-mapping>`. This check
+runs only if external networks IDs values are specified in the mapping file
+(``[migrate] ext_net_map`` config option contains path to this file). It
+verifies existence and validity of external networks (which sometimes referred
+as public or floating networks), specified in the map file.
+
+It simply runs ``neutron net-show <net-id>`` command and verifies that the
+networks, which are specified in the mapping file are actually exist and theirs
+IDs are valid.
+
+
+External networks overlap
+-------------------------
+
+Checks :ref:`external-networks-overlap`. Check starts only if
+``[migrate] migrate_ext_nets`` config option is set to ``True`` in the
+:ref:`primary-config-file`. This check verifies if there is any IP range
+overlapping between source and destination clouds' external networks. In
+particular, external subnets are compared. The main idea is to check if theirs
+allocation pools overlap. Only in this case migration will not start. If
+subnets have the same CIDRs, but different allocation pools, migration will
+proceed, because in this case there are no any IP conflicts. Example::
```

Comment:
Please add description on why this is important and what types of errors external networks overlap can lead to.